sidecar/src/service/tracing/trace_flusher.rs

```diff
@@ -200,12 +214,63 @@ impl TraceFlusher {
         }
     }
 
+    fn replace_trace_send_data(&self, completer: ManualFutureCompleter<()>) -> Vec<SendData> {
```

Comment:
No logic changes here, just breaking up the `start_trace_flusher` function into smaller functions.